qml/TabLayout.qml

```diff
@@ -248,37 +257,149 @@ Item {
+                                    {
+                                        domain_id_dialog.open()
                                     }
-                                    stack.push(domainViewLayout)
-                                    refresh_layout(current_)
                                 }
                             }
                         }
                     }
                 }
+
+                Component {
+                    id: domainGraphLayout_component
+
+                    DomainGraphLayout
+                    {
+                        id: domainGraphLayout
+                        component_id: stack.stack_id
+
+                        onUpdate_tab_name: {
+                            tabLayout.tab_model_[current_]["title"] = new_name
+                            refresh_layout(current_)
+                        }
+
+                        Connections {
+                            target: tabLayout
+
+                            function onInitialize_domain_view_(stack_id, entity_id, domain_id) {
+                                if (domainGraphLayout.component_id == stack_id)
+                                {
+                                    domainGraphLayout.entity_id = entity_id
+                                    domainGraphLayout.domain_id = domain_id
+                                    domainGraphLayout.load_model()
+                                }
+                            }
+                        }
+
+                    }
+                }
+
+                Connections {
+                    target: tabLayout
+
+                    function onOpen_domain_view_(stack_id, entity_id, domain_id) {
+                        if (stack.stack_id == stack_id)
+                        {
+                            if (stack.deep > 1)
+                            {
+                                stack.pop()
+                            }
+
+                            stack.push(domainGraphLayout_component)
+                            refresh_layout(current_)
+                            initialize_domain_view_(stack_id, entity_id, domain_id)
+                        }
+                    }
+                }
             }
         }
     }
 
-    Component {
-        id: domainViewLayout
+    Dialog {
+        id: domain_id_dialog
+
+        property bool enable_ok_button: false       // disable OK button until user selects domain id
 
-        Rectangle{
-            Text{
-                text: "Here would be the domain view"
+        x: (parent.width - width) / 2
+        y: (parent.height - height) / 2
+
+        width: 300
+
+        modal: true
+        title: "Select DDS Domain"
+
+        footer: DialogButtonBox {
+            id: buttons
+            standardButtons: Dialog.Ok | Dialog.Cancel
+        }
+
+        Component.onCompleted: {
+            controller.update_available_entity_ids("Domain", "getDataDialogSourceEntityId")
+        }
```

Comment:
`update_available_entity_ids()` call should only be called in `onAboutToShow` callback. Also delete this call in `onCompleted` callback in AdaptativeComboBox.